src/awkward/_v2/_lookup.py

```diff
@@ -81,6 +81,22 @@ def tolookup_identifier(cls, layout, positions):
             positions.append(layout.identifier.data)
 
     def IndexOf(self, arraytype):
+        if isinstance(arraytype, str):
+            if arraytype == "int8_t":
+                return ak._v2.index.Index8
+            elif arraytype == "uint8_t":
+                return ak._v2.index.IndexU8
+            elif arraytype == "int32_t":
+                return ak._v2.index.Index32
+            elif arraytype == "uint32_t":
+                return ak._v2.index.IndexU32
+            elif arraytype == "int64_t":
+                return ak._v2.index.Index64
+            elif arraytype == "uint64_t":
+                return ak._v2.index.IndexU64
+            else:
+                raise ak._v2._util.error(AssertionError(arraytype))
+
         if arraytype.dtype.bitwidth == 8 and arraytype.dtype.signed:
             return ak._v2.index.Index8
         elif arraytype.dtype.bitwidth == 8:
```

Comment:
If this Lookup is subclassed for C++, the `arraytype` is a string, and if this Lookup is subclassed for Numba, it's a `nb.Type`. Specializing the behavior with an if-then in the superclass is breaking some rules of type inheritance.

What should be happening here is that the Lookup superclass (in src/awkward/_v2/_lookup.py) shouldn't have an `IndexOf` method; an `IndexOf` for C++ (`arraytype` is a string) should be mixed into the Generators in cling.py and an `IndexOf` for Numba (`arraytype` is an `nb.Type`) should be mixed into the Lookups in _connect/numba/arrayview.py.

It means touching the Numba code (to move the original `IndexOf` into _connect/numba/arrayview.py), but that's better than having this historical mistake compounded.

Reply:
done

Reply:
I've checked locally, all Numba tests pass except for the two that require `cppyy.numba_ext`. My local `cppyy` v 1.6.2 comes from ROOT local build that is rather old. The Numba support `cppyy` module is available from 2.4.0.

Reply:
This change, here, addresses Numba independently of cppyy. Are you saying that having cppyy installed (with its Numba modifications) breaks some of the Numba tests? If so, that's something Wim will need to know.

Reply:
```zsh
python -m pytest -vv -rs tests/v2/test_1240-v2-implementation-of-numba-1.py
========================================================================================= test session starts =========================================================================================
platform darwin -- Python 3.10.1, pytest-6.2.5, py-1.11.0, pluggy-1.0.0 -- /Users/yana/.pyenv/versions/3.10.1/bin/python
cachedir: .pytest_cache
rootdir: /Users/yana/Projects/PR1613/awkward, configfile: pyproject.toml
plugins: anyio-3.5.0, cov-3.0.0
collected 21 items                                                                                                                                                                                    

tests/v2/test_1240-v2-implementation-of-numba-1.py::test_EmptyArray FAILED                                                                                                                      [  4%]
tests/v2/test_1240-v2-implementation-of-numba-1.py::test_NumpyArray PASSED                                                                                                                      [  9%]
tests/v2/test_1240-v2-implementation-of-numba-1.py::test_RegularArray_NumpyArray PASSED                                                                                                         [ 14%]
tests/v2/test_1240-v2-implementation-of-numba-1.py::test_ListArray_NumpyArray PASSED                                                                                                            [ 19%]
tests/v2/test_1240-v2-implementation-of-numba-1.py::test_ListOffsetArray_NumpyArray PASSED                                                                                                      [ 23%]
tests/v2/test_1240-v2-implementation-of-numba-1.py::test_RecordArray_NumpyArray PASSED                                                                                                          [ 28%]
tests/v2/test_1240-v2-implementation-of-numba-1.py::test_IndexedArray_NumpyArray PASSED                                                                                                         [ 33%]
tests/v2/test_1240-v2-implementation-of-numba-1.py::test_IndexedOptionArray_NumpyArray PASSED                                                                                                   [ 38%]
tests/v2/test_1240-v2-implementation-of-numba-1.py::test_ByteMaskedArray_NumpyArray PASSED                                                                                                      [ 42%]
tests/v2/test_1240-v2-implementation-of-numba-1.py::test_BitMaskedArray_NumpyArray PASSED                                                                                                       [ 47%]
tests/v2/test_1240-v2-implementation-of-numba-1.py::test_UnmaskedArray_NumpyArray PASSED                                                                                                        [ 52%]
tests/v2/test_1240-v2-implementation-of-numba-1.py::test_UnionArray_NumpyArray PASSED                                                                                                           [ 57%]
tests/v2/test_1240-v2-implementation-of-numba-1.py::test_RegularArray_RecordArray_NumpyArray PASSED                                                                                             [ 61%]
tests/v2/test_1240-v2-implementation-of-numba-1.py::test_ListArray_RecordArray_NumpyArray PASSED                                                                                                [ 66%]
tests/v2/test_1240-v2-implementation-of-numba-1.py::test_ListOffsetArray_RecordArray_NumpyArray PASSED                                                                                          [ 71%]
tests/v2/test_1240-v2-implementation-of-numba-1.py::test_IndexedArray_RecordArray_NumpyArray PASSED                                                                                             [ 76%]
tests/v2/test_1240-v2-implementation-of-numba-1.py::test_IndexedOptionArray_RecordArray_NumpyArray PASSED                                                                                       [ 80%]
tests/v2/test_1240-v2-implementation-of-numba-1.py::test_ByteMaskedArray_RecordArray_NumpyArray PASSED                                                                                          [ 85%]
tests/v2/test_1240-v2-implementation-of-numba-1.py::test_BitMaskedArray_RecordArray_NumpyArray PASSED                                                                                           [ 90%]
tests/v2/test_1240-v2-implementation-of-numba-1.py::test_UnmaskedArray_RecordArray_NumpyArray PASSED                                                                                            [ 95%]
tests/v2/test_1240-v2-implementation-of-numba-1.py::test_UnionArray_RecordArray_NumpyArray PASSED                                                                                               [100%]

============================================================================================== FAILURES ===============================================================================================
___________________________________________________________________________________________ test_EmptyArray ___________________________________________________________________________________________

entry_point = EntryPoint(name='init', value='cppyy.numba_ext:_init_extension', group='numba_extensions')

    def load_ep(entry_point):
        """Loads a given entry point. Warns and logs on failure.
        """
        logger.debug('Loading extension: %s', entry_point)
        try:
>           func = entry_point.load()

entry_point = EntryPoint(name='init', value='cppyy.numba_ext:_init_extension', group='numba_extensions')
msg        = ("Numba extension module 'cppyy.numba_ext' failed to load due to "
 "'ModuleNotFoundError(No module named 'cppyy.numba_ext')'.")

../../../.pyenv/versions/3.10.1/lib/python3.10/site-packages/numba/core/entrypoints.py:40: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = EntryPoint(name='init', value='cppyy.numba_ext:_init_extension', group='numba_extensions')

    def load(self):
        """Load the entry point from its definition. If only a module
        is indicated by the value, return that module. Otherwise,
        return the named object.
        """
        match = self.pattern.match(self.value)
>       module = import_module(match.group('module'))

match      = <re.Match object; span=(0, 31), match='cppyy.numba_ext:_init_extension'>
self       = EntryPoint(name='init', value='cppyy.numba_ext:_init_extension', group='numba_extensions')

../../../.pyenv/versions/3.10.1/lib/python3.10/importlib/metadata/__init__.py:162: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

name = 'cppyy.numba_ext', package = None

    def import_module(name, package=None):
        """Import a module.
    
        The 'package' argument is required when performing a relative import. It
        specifies the package to use as the anchor point from which to resolve the
        relative import to an absolute import.
    
        """
        level = 0
        if name.startswith('.'):
            if not package:
                msg = ("the 'package' argument is required to perform a relative "
                       "import for {!r}")
                raise TypeError(msg.format(name))
            for character in name:
                if character != '.':
                    break
                level += 1
>       return _bootstrap._gcd_import(name[level:], package, level)

level      = 0
name       = 'cppyy.numba_ext'
package    = None

../../../.pyenv/versions/3.10.1/lib/python3.10/importlib/__init__.py:126: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

name = 'cppyy.numba_ext', package = None, level = 0

>   ???

level      = 0
name       = 'cppyy.numba_ext'
package    = None

<frozen importlib._bootstrap>:1050: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

name = 'cppyy.numba_ext', import_ = <function _gcd_import at 0x10322b400>

>   ???

import_    = <function _gcd_import at 0x10322b400>
module     = <object object at 0x10324c060>
name       = 'cppyy.numba_ext'

<frozen importlib._bootstrap>:1027: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

name = 'cppyy.numba_ext', import_ = <function _gcd_import at 0x10322b400>

>   ???
E   ModuleNotFoundError: No module named 'cppyy.numba_ext'

import_    = <function _gcd_import at 0x10322b400>
name       = 'cppyy.numba_ext'
parent     = 'cppyy'
parent_module = <module 'cppyy' from '/Users/yana/Projects/ROOT/ROOT-master/21.04.2022/root_install/lib/cppyy/__init__.py'>
path       = ['/Users/yana/Projects/ROOT/ROOT-master/21.04.2022/root_install/lib/cppyy']
spec       = None

<frozen importlib._bootstrap>:1004: ModuleNotFoundError

During handling of the above exception, another exception occurred:

    def test_EmptyArray():
        v2a = ak._v2.contents.emptyarray.EmptyArray().toNumpyArray(np.dtype(np.float64))
        roundtrip(v2a)
        array = ak._v2.highlevel.Array(v2a)
>       memoryleak(array, swallow)

array      = <Array [] type='0 * float64'>
v2a        = <NumpyArray dtype='float64' len='0'>[]</NumpyArray>

tests/v2/test_1240-v2-implementation-of-numba-1.py:88: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
tests/v2/test_1240-v2-implementation-of-numba-1.py:76: in memoryleak
    function(array)
        _          = 0
        array      = <Array [] type='0 * float64'>
        counts     = None
        function   = CPUDispatcher(<function swallow at 0x1208e2a70>)
../../../.pyenv/versions/3.10.1/lib/python3.10/site-packages/numba/core/dispatcher.py:396: in _compile_for_args
    self._compilation_chain_init_hook()
        args       = (<Array [] type='0 * float64'>,)
        kws        = {}
        self       = CPUDispatcher(<function swallow at 0x1208e2a70>)
../../../.pyenv/versions/3.10.1/lib/python3.10/site-packages/numba/core/dispatcher.py:289: in _compilation_chain_init_hook
    entrypoints.init_all()
        self       = CPUDispatcher(<function swallow at 0x1208e2a70>)
../../../.pyenv/versions/3.10.1/lib/python3.10/site-packages/numba/core/entrypoints.py:54: in init_all
    load_ep(entry_point)
        entry_point = EntryPoint(name='init', value='cppyy.numba_ext:_init_extension', group='numba_extensions')
        eps        = {'array_api': [EntryPoint(name='numpy', value='numpy.array_api', group='array_api')],
 'babel.checkers': [EntryPoint(name='num_plurals', value='babel.messages.checkers:num_plurals', group='babel.checkers'),
                    EntryPoint(name='python_format', value='babel.messages.checkers:python_format', group='babel.checkers')],
 'babel.extractors': [EntryPoint(name='jinja2', value='jinja2.ext:babel_extract[i18n]', group='babel.extractors'),
                      EntryPoint(name='ignore', value='babel.messages.extract:extract_nothing', group='babel.extractors'),
                      EntryPoint(name='javascript', value='babel.messages.extract:extract_javascript', group='babel.extractors'),
                      EntryPoint(name='python', value='babel.messages.extract:extract_python', group='babel.extractors')],
 'console_scripts': [EntryPoint(name='ipython', value='IPython:start_ipython', group='console_scripts'),
                     EntryPoint(name='ipython3', value='IPython:start_ipython', group='console_scripts'),
                     EntryPoint(name='cling-config', value='cppyy_backend._cling_config:main', group='console_scripts'),
                     EntryPoint(name='cppyy-generator', value='cppyy_backend._cppyy_generator:main', group='console_scripts'),
                     EntryPoint(name='genreflex', value='cppyy_backend._genreflex:main', group='console_scripts'),
                     EntryPoint(name='rootcling', value='cppyy_backend._rootcling:main', group='console_scripts'),
                     EntryPoint(name='jupyter-kernel', value='jupyter_client.kernelapp:main', group='console_scripts'),
                     EntryPoint(name='jupyter-kernelspec', value='jupyter_client.kernelspecapp:KernelSpecApp.launch_instance', group='console_scripts'),
                     EntryPoint(name='jupyter-run', value='jupyter_client.runapp:RunApp.launch_instance', group='console_scripts'),
                     EntryPoint(name='coverage', value='coverage.cmdline:main', group='console_scripts'),
                     EntryPoint(name='coverage-3.10', value='coverage.cmdline:main', group='console_scripts'),
                     EntryPoint(name='coverage3', value='coverage.cmdline:main', group='console_scripts'),
                     EntryPoint(name='send2trash', value='send2trash.__main__:main', group='console_scripts'),
                     EntryPoint(name='f2py', value='numpy.f2py.f2py2e:main', group='console_scripts'),
                     EntryPoint(name='f2py3', value='numpy.f2py.f2py2e:main', group='console_scripts'),
                     EntryPoint(name='f2py3.10', value='numpy.f2py.f2py2e:main', group='console_scripts'),
                     EntryPoint(name='jupyter-dejavu', value='nbconvert.nbconvertapp:dejavu_main', group='console_scripts'),
                     EntryPoint(name='jupyter-nbconvert', value='nbconvert.nbconvertapp:main', group='console_scripts'),
                     EntryPoint(name='jupyter-nbclassic', value='nbclassic.notebookapp:main', group='console_scripts'),
                     EntryPoint(name='kernprof', value='kernprof:main', group='console_scripts'),
                     EntryPoint(name='pygmentize', value='pygments.cmdline:main', group='console_scripts'),
                     EntryPoint(name='jupyter-execute', value='nbclient.cli:main', group='console_scripts'),
                     EntryPoint(name='jupyter', value='jupyter_core.command:main', group='console_scripts'),
                     EntryPoint(name='jupyter-migrate', value='jupyter_core.migrate:main', group='console_scripts'),
                     EntryPoint(name='jupyter-troubleshoot', value='jupyter_core.troubleshoot:main', group='console_scripts'),
                     EntryPoint(name='wsdump', value='websocket._wsdump:main', group='console_scripts'),
                     EntryPoint(name='pybabel', value='babel.messages.frontend:main', group='console_scripts'),
                     EntryPoint(name='jlpm', value='jupyterlab.jlpmapp:main', group='console_scripts'),
                     EntryPoint(name='jupyter-lab', value='jupyterlab.labapp:main', group='console_scripts'),
                     EntryPoint(name='jupyter-labextension', value='jupyterlab.labextensions:main', group='console_scripts'),
                     EntryPoint(name='jupyter-labhub', value='jupyterlab.labhubapp:main', group='console_scripts'),
                     EntryPoint(name='pip', value='pip._internal.cli.main:main', group='console_scripts'),
                     EntryPoint(name='pip3', value='pip._internal.cli.main:main', group='console_scripts'),
                     EntryPoint(name='pip3.8', value='pip._internal.cli.main:main', group='console_scripts'),
                     EntryPoint(name='black', value='black:patched_main', group='console_scripts'),
                     EntryPoint(name='black-primer', value='black_primer.cli:main', group='console_scripts'),
                     EntryPoint(name='blackd', value='blackd:patched_main [d]', group='console_scripts'),
                     EntryPoint(name='py.test', value='pytest:console_main', group='console_scripts'),
                     EntryPoint(name='pytest', value='pytest:console_main', group='console_scripts'),
                     EntryPoint(name='jupyter-trust', value='nbformat.sign:TrustNotebookApp.launch_instance', group='console_scripts'),
                     EntryPoint(name='normalizer', value='charset_normalizer.cli.normalizer:cli_detect', group='console_scripts'),
                     EntryPoint(name='pyjson5', value='json5.tool:main', group='console_scripts'),
                     EntryPoint(name='jsonschema', value='jsonschema.cli:main', group='console_scripts'),
                     EntryPoint(name='jupyter-server', value='jupyter_server.serverapp:main', group='console_scripts'),
                     EntryPoint(name='jupyter-bundlerextension', value='notebook.bundler.bundlerextensions:main', group='console_scripts'),
                     EntryPoint(name='jupyter-nbextension', value='notebook.nbextensions:main', group='console_scripts'),
                     EntryPoint(name='jupyter-notebook', value='notebook.notebookapp:main', group='console_scripts'),
                     EntryPoint(name='jupyter-serverextension', value='notebook.serverextensions:main', group='console_scripts')],
 'distutils.commands': [EntryPoint(name='compile_catalog', value='babel.messages.frontend:compile_catalog', group='distutils.commands'),
                        EntryPoint(name='extract_messages', value='babel.messages.frontend:extract_messages', group='distutils.commands'),
                        EntryPoint(name='init_catalog', value='babel.messages.frontend:init_catalog', group='distutils.commands'),
                        EntryPoint(name='update_catalog', value='babel.messages.frontend:update_catalog', group='distutils.commands'),
                        EntryPoint(name='alias', value='setuptools.command.alias:alias', group='distutils.commands'),
                        EntryPoint(name='bdist_egg', value='setuptools.command.bdist_egg:bdist_egg', group='distutils.commands'),
                        EntryPoint(name='bdist_rpm', value='setuptools.command.bdist_rpm:bdist_rpm', group='distutils.commands'),
                        EntryPoint(name='build_clib', value='setuptools.command.build_clib:build_clib', group='distutils.commands'),
                        EntryPoint(name='build_ext', value='setuptools.command.build_ext:build_ext', group='distutils.commands'),
                        EntryPoint(name='build_py', value='setuptools.command.build_py:build_py', group='distutils.commands'),
                        EntryPoint(name='develop', value='setuptools.command.develop:develop', group='distutils.commands'),
                        EntryPoint(name='dist_info', value='setuptools.command.dist_info:dist_info', group='distutils.commands'),
                        EntryPoint(name='easy_install', value='setuptools.command.easy_install:easy_install', group='distutils.commands'),
                        EntryPoint(name='egg_info', value='setuptools.command.egg_info:egg_info', group='distutils.commands'),
                        EntryPoint(name='install', value='setuptools.command.install:install', group='distutils.commands'),
                        EntryPoint(name='install_egg_info', value='setuptools.command.install_egg_info:install_egg_info', group='distutils.commands'),
                        EntryPoint(name='install_lib', value='setuptools.command.install_lib:install_lib', group='distutils.commands'),
                        EntryPoint(name='install_scripts', value='setuptools.command.install_scripts:install_scripts', group='distutils.commands'),
                        EntryPoint(name='rotate', value='setuptools.command.rotate:rotate', group='distutils.commands'),
                        EntryPoint(name='saveopts', value='setuptools.command.saveopts:saveopts', group='distutils.commands'),
                        EntryPoint(name='sdist', value='setuptools.command.sdist:sdist', group='distutils.commands'),
                        EntryPoint(name='setopt', value='setuptools.command.setopt:setopt', group='distutils.commands'),
                        EntryPoint(name='test', value='setuptools.command.test:test', group='distutils.commands'),
                        EntryPoint(name='upload_docs', value='setuptools.command.upload_docs:upload_docs', group='distutils.commands')],
 'distutils.setup_keywords': [EntryPoint(name='message_extractors', value='babel.messages.frontend:check_message_extractors', group='distutils.setup_keywords'),
                              EntryPoint(name='cffi_modules', value='cffi.setuptools_ext:cffi_modules', group='distutils.setup_keywords'),
                              EntryPoint(name='dependency_links', value='setuptools.dist:assert_string_list', group='distutils.setup_keywords'),
                              EntryPoint(name='eager_resources', value='setuptools.dist:assert_string_list', group='distutils.setup_keywords'),
                              EntryPoint(name='entry_points', value='setuptools.dist:check_entry_points', group='distutils.setup_keywords'),
                              EntryPoint(name='exclude_package_data', value='setuptools.dist:check_package_data', group='distutils.setup_keywords'),
                              EntryPoint(name='extras_require', value='setuptools.dist:check_extras', group='distutils.setup_keywords'),
                              EntryPoint(name='include_package_data', value='setuptools.dist:assert_bool', group='distutils.setup_keywords'),
                              EntryPoint(name='install_requires', value='setuptools.dist:check_requirements', group='distutils.setup_keywords'),
                              EntryPoint(name='namespace_packages', value='setuptools.dist:check_nsp', group='distutils.setup_keywords'),
                              EntryPoint(name='package_data', value='setuptools.dist:check_package_data', group='distutils.setup_keywords'),
                              EntryPoint(name='packages', value='setuptools.dist:check_packages', group='distutils.setup_keywords'),
                              EntryPoint(name='python_requires', value='setuptools.dist:check_specifier', group='distutils.setup_keywords'),
                              EntryPoint(name='setup_requires', value='setuptools.dist:check_requirements', group='distutils.setup_keywords'),
                              EntryPoint(name='test_loader', value='setuptools.dist:check_importable', group='distutils.setup_keywords'),
                              EntryPoint(name='test_runner', value='setuptools.dist:check_importable', group='distutils.setup_keywords'),
                              EntryPoint(name='test_suite', value='setuptools.dist:check_test_suite', group='distutils.setup_keywords'),
                              EntryPoint(name='tests_require', value='setuptools.dist:check_requirements', group='distutils.setup_keywords'),
                              EntryPoint(name='use_2to3', value='setuptools.dist:invalid_unless_false', group='distutils.setup_keywords'),
                              EntryPoint(name='zip_safe', value='setuptools.dist:assert_bool', group='distutils.setup_keywords')],
 'egg_info.writers': [EntryPoint(name='PKG-INFO', value='setuptools.command.egg_info:write_pkg_info', group='egg_info.writers'),
                      EntryPoint(name='dependency_links.txt', value='setuptools.command.egg_info:overwrite_arg', group='egg_info.writers'),
                      EntryPoint(name='depends.txt', value='setuptools.command.egg_info:warn_depends_obsolete', group='egg_info.writers'),
                      EntryPoint(name='eager_resources.txt', value='setuptools.command.egg_info:overwrite_arg', group='egg_info.writers'),
                      EntryPoint(name='entry_points.txt', value='setuptools.command.egg_info:write_entries', group='egg_info.writers'),
                      EntryPoint(name='namespace_packages.txt', value='setuptools.command.egg_info:overwrite_arg', group='egg_info.writers'),
                      EntryPoint(name='requires.txt', value='setuptools.command.egg_info:write_requirements', group='egg_info.writers'),
                      EntryPoint(name='top_level.txt', value='setuptools.command.egg_info:write_toplevel_names', group='egg_info.writers')],
 'jupyter_client.kernel_provisioners': [EntryPoint(name='local-provisioner', value='jupyter_client.provisioning:LocalProvisioner', group='jupyter_client.kernel_provisioners')],
 'nbconvert.exporters': [EntryPoint(name='asciidoc', value='nbconvert.exporters:ASCIIDocExporter', group='nbconvert.exporters'),
                         EntryPoint(name='custom', value='nbconvert.exporters:TemplateExporter', group='nbconvert.exporters'),
                         EntryPoint(name='html', value='nbconvert.exporters:HTMLExporter', group='nbconvert.exporters'),
                         EntryPoint(name='latex', value='nbconvert.exporters:LatexExporter', group='nbconvert.exporters'),
                         EntryPoint(name='markdown', value='nbconvert.exporters:MarkdownExporter', group='nbconvert.exporters'),
                         EntryPoint(name='notebook', value='nbconvert.exporters:NotebookExporter', group='nbconvert.exporters'),
                         EntryPoint(name='pdf', value='nbconvert.exporters:PDFExporter', group='nbconvert.exporters'),
                         EntryPoint(name='python', value='nbconvert.exporters:PythonExporter', group='nbconvert.exporters'),
                         EntryPoint(name='rst', value='nbconvert.exporters:RSTExporter', group='nbconvert.exporters'),
                         EntryPoint(name='script', value='nbconvert.exporters:ScriptExporter', group='nbconvert.exporters'),
                         EntryPoint(name='slides', value='nbconvert.exporters:SlidesExporter', group='nbconvert.exporters'),
                         EntryPoint(name='webpdf', value='nbconvert.exporters:WebPDFExporter', group='nbconvert.exporters')],
 'numba_extensions': [EntryPoint(name='init', value='cppyy.numba_ext:_init_extension', group='numba_extensions')],
 'pygments.lexers': [EntryPoint(name='ipython', value='IPython.lib.lexers:IPythonLexer', group='pygments.lexers'),
                     EntryPoint(name='ipython3', value='IPython.lib.lexers:IPython3Lexer', group='pygments.lexers'),
                     EntryPoint(name='ipythonconsole', value='IPython.lib.lexers:IPythonConsoleLexer', group='pygments.lexers')],
 'pytest11': [EntryPoint(name='anyio', value='anyio.pytest_plugin', group='pytest11'),
              EntryPoint(name='pytest_cov', value='pytest_cov.plugin', group='pytest11')],
 'setuptools.finalize_distribution_options': [EntryPoint(name='keywords', value='setuptools.dist:Distribution._finalize_setup_keywords', group='setuptools.finalize_distribution_options'),
                                              EntryPoint(name='parent_finalize', value='setuptools.dist:_Distribution.finalize_options', group='setuptools.finalize_distribution_options')]}
        load_ep    = <function init_all.<locals>.load_ep at 0x12097b880>
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

entry_point = EntryPoint(name='init', value='cppyy.numba_ext:_init_extension', group='numba_extensions')

    def load_ep(entry_point):
        """Loads a given entry point. Warns and logs on failure.
        """
        logger.debug('Loading extension: %s', entry_point)
        try:
            func = entry_point.load()
            func()
        except Exception as e:
            msg = (f"Numba extension module '{entry_point.module}' "
                   f"failed to load due to '{type(e).__name__}({str(e)})'.")
>           warnings.warn(msg, stacklevel=3)
E           UserWarning: Numba extension module 'cppyy.numba_ext' failed to load due to 'ModuleNotFoundError(No module named 'cppyy.numba_ext')'.

entry_point = EntryPoint(name='init', value='cppyy.numba_ext:_init_extension', group='numba_extensions')
msg        = ("Numba extension module 'cppyy.numba_ext' failed to load due to "
 "'ModuleNotFoundError(No module named 'cppyy.numba_ext')'.")

../../../.pyenv/versions/3.10.1/lib/python3.10/site-packages/numba/core/entrypoints.py:45: UserWarning
==================================================================================== 1 failed, 20 passed in 13.16s ====================================================================================
```

Reply:
`cppyy.numba_ext:_init_extension` (cppyy's Numba extension-point) is breaking "`import numba`". (https://github.com/wlav/cppyy)